src/withResolution/mediaQuery.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+type Callback = (mql: MediaQueryList, alias: string) => void
+type QueryData = {
+  [query: string]: {|
+    +mediaQuery: MediaQueryList,
+    +allCallbacks: (mql: MediaQueryListEvent) => void,
+  |},
+}
+type PerCallbackData = {
+  [query: string]: Array<{|
+    +alias: string,
+    +callback: Callback,
+  |}>,
+}
```

Comment:
media queries manages setting up the minimal number of listeners possible and removing them when they are no longer in use, based on some quick jsfiddles this roughly doubles performance on Chrome